OutOfSchool/OutOfSchool.WebApi/Controllers/V1/AdminController.cs

```diff
@@ -1,44 +1,274 @@
-using AutoMapper;
+using Google.Apis.Discovery;
+using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Localization;
+using Microsoft.FeatureManagement.Mvc;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.WebApi.Common;
+using OutOfSchool.WebApi.Enums;
 using OutOfSchool.WebApi.Models;
+using OutOfSchool.WebApi.Models.Application;
+using OutOfSchool.WebApi.Models.Providers;
 
 namespace OutOfSchool.WebApi.Controllers.V1;
 
 [ApiController]
 [ApiVersion("1.0")]
 [Route("api/v{version:apiVersion}/[controller]/[action]")]
-public class AdminController : ControllerBase
+public class AdminController : Controller
 {
-    private readonly IUserService userService;
-    private readonly IMapper mapper;
+    private readonly IStringLocalizer<SharedResource> localizer;
 
-    public AdminController(IUserService userService, IMapper mapper)
+    private readonly ILogger<AdminController> logger;
+
+    private readonly ICurrentUserService currentUserService;
+    private readonly ISensitiveMinistryAdminService ministryAdminService;
+    private readonly ISensitiveDirectionService directionService;
+    private readonly ISensitiveProviderService providerService;
+    private readonly ISensitiveApplicationService applicationService;
+
+
+    public AdminController(
+       ILogger<AdminController> logger,
+       ICurrentUserService currentUserService,
+       ISensitiveMinistryAdminService ministryAdminService,
+       ISensitiveApplicationService applicationService,
+       ISensitiveDirectionService directionService,
+       ISensitiveProviderService providerService,
+       IStringLocalizer<SharedResource> localizer)
+    {
+        this.localizer = localizer;
+        this.logger = logger;
+        this.applicationService = applicationService;
+        this.directionService = directionService;
+        this.providerService = providerService;
+        this.currentUserService = currentUserService ?? throw new ArgumentNullException(nameof(currentUserService));
```

Comment:
> @ZdorovenkoKyrylo Is it pushed to server? Can't find the change

I hope it will be pushed today together with implementation your comment about test below. 
Thank you for it)
I just highlight for myself to find comments which I should do. Because there so many of them. I used practice resolving before. Now I follow your tips and just write the comment "Done"